src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -57,63 +97,155 @@ ActorInit En_Door_InitVars = {
     /**/ EnDoor_Draw,
 };
 
-typedef struct {
+typedef enum EnDoorDListIndex {
+    /*  0 */ DOOR_DL_DEFAULT,
+    /*  1 */ DOOR_DL_WOODFALL,
+    /*  2 */ DOOR_DL_OBSERVATORY_LAB,
+    /*  3 */ DOOR_DL_ZORA_HALL,
+    /*  4 */ DOOR_DL_SWAMP,
+    /*  5 */ DOOR_DL_MAGIC_HAG_POTION_SHOP,
+    /*  6 */ DOOR_DL_LOTTERY_CURIOSITY_SHIP_MAYOR_HOUSE,
+    /*  7 */ DOOR_DL_POST_OFFICE_TRAIDING_POST,
+    /*  8 */ DOOR_DL_INN_SCHOOL,
+    /*  9 */ DOOR_DL_MILK_BAR,
+    /* 10 */ DOOR_DL_MUSIC_BOX,
+    /* 11 */ DOOR_DL_PIRATES_FORTESS,
+    /* 12 */ DOOR_DL_OCEANSIDE_SPIDER_HOUSE,
+    /* 13 */ DOOR_DL_DEFAULT_FIELD_KEEP,
+    /* 14 */ DOOR_DL_MAX
+} EnDoorDListIndex;
+
+typedef struct EnDoorInfo {
     /* 0x0 */ s16 sceneId;
     /* 0x2 */ u8 dListIndex;
     /* 0x4 */ s16 objectId;
 } EnDoorInfo; // size = 0x6
 
+// TODO: This enum feels kinda overkill...
+typedef enum EnDoorObjectInfoIndex {
+    /*  0 */ DOOR_OBJKIND_DEFAULT,
+    /*  0 */ DOOR_OBJINFO_0 = DOOR_OBJKIND_DEFAULT,
+    /*  1 */ DOOR_OBJINFO_1,
+    /*  2 */ DOOR_OBJINFO_2,
+    /*  3 */ DOOR_OBJINFO_3,
```

Comment:
I'm not completely sure about this enum.
I initially thought it would be like Anju's animation enum because it also has ranges where each range has its own meaning, but most values of this enum are not explicitly referenced in the code, so I'm not really sure if it is worth having.
At the same time, having it and keeping the `static_assert`s give a bit extra safety on the code's shiftability.